src/riva/asrlib/decoder/scripts/prepare_TLG_fst/local/prepare_dict_ctc.sh

```diff
@@ -112,8 +116,20 @@ if [ $probabilistic = true ]; then
     echo "probabilistic lexicon filtering is not supported"
     cut -d" " -f1,3- $dir/lexiconp.txt > $dir/lexicon.txt
 else
-    (echo '<spoken_noise> [UNK]'; echo '<unk> [UNK]'; ) | cat - $lexicon | sort | uniq > $dir/lexicon.tmp || exit 1;
-    grep -vFf <(comm -23 <(cut -d" " -f2- $dir/lexicon.tmp | tr ' ' '\n' | sort -u) <(sort $dir/units.txt)) $dir/lexicon.tmp > $dir/lexicon.txt
+    # Do not understand. Too hard to read!!!!
+    (echo '<spoken_noise> [UNK]'; echo '<unk> [UNK]'; ) | cat - $lexicon | sed 's/\t/ /g' | sort | uniq > $dir/lexicon.tmp || exit 1;
+    # It has a problem when the lexicon does not fully cover the units, it seems...
+    cut -d" " -f2- $dir/lexicon.tmp > $dir/1.txt
+    cut -d" " -f2- $dir/lexicon.tmp | tr ' ' '\n' > $dir/2.txt
+    cut -d" " -f2- $dir/lexicon.tmp | tr ' ' '\n' | sort -u > $dir/units_from_lexicon_sorted.txt
+    sort $dir/units.txt > $dir/units_sorted.txt
+    cmp $dir/units_sorted.txt $dir/units_from_lexicon_sorted.txt
+    if [ $? ]; then
+        echo "ERROR: Difference in units.txt and units derived from lexicon!"
```

Comment:
This seems to be a false positive when it triggers. It requires more inspection.